docs/adrs/00005-license-export-analytics.md

```diff
@@ -0,0 +1,368 @@
+           .column_as(license::Column::SpdxLicenses, "spdx_licenses")
+           .column_as(
+              license::Column::SpdxLicenseExceptions,
+              "spdx_license_exceptions",
+           )
+           .into_model::<SbomLicenseBase>()
+           .all(&ctx.db)
+           .await?;
+  
+   Ok(())
+}
+```
+As shown in the above example, we are currently storing both licenseDeclared and licenseConcluded, and we are unable to distinguish between them.
+At same time, if the SBOM is cycloneDX, component_group will miss.The information in the ExtractedLicensingInfos section is completely missing.
+
+## Decision
+
+### Current design proposal
+#### Data model
+The current design requires adding a license_type field in purl_license_assertion, as well as a new entity called extracted_licensing_infos. Additionally, Sbom_node Entity add new field group.
+![erd.png](erd.png)
+```rust
+pub enum LicenseCategory {
+    #[sea_orm(string_value = "slc")]
+    SpdxDeclared,
+    #[sea_orm(string_value = "sld")]
+    SpdxConcluded,
+    #[sea_orm(string_value = "clci")]
+    CydxLcid,
+    #[sea_orm(string_value = "clcn")]
+    CydxLcname,
+    #[sea_orm(string_value = "cle")]
+    CydxLexpression,
+    #[sea_orm(string_value = "cd")]
+    ClearlyDefined,
+    #[sea_orm(string_value = "o")]
+    Other,
+}
+```
+
+Currently, the system's licenses come from the following sources: 1) SPDX's declared license and concluded license, 2) CycloneDX, and 3) clearlyDefined. The license export only requires SPDX's declared license and CycloneDX licenses, while concluded licenses and clearlyDefined licenses are not needed. Therefore, the system must distinguish between them.
+
+SpdxDeclared: The license from spdx sbom's declered license.
+SpdxConcluded: The license from spdx sbom's concluded license.
+CydxLcid: The license from CycloneDX sbom's id.
+CydxLcname: The license from CycloneDX sbom's name.
+CydxLexpression: The license from CycloneDX sbom's Expression.
+
+#### SBOM ingestion
+Add the following actions during the ingestion process.
+1. add component_group into Entity sbom_node.
+2. add LicenseCategory into Entity purl_license_assertion.
+3. create new Entity extracted_licensing_infos.
+
+#### License Extract API
+This section of the code can be divided into three parts.
+1. Data Collection
+   - This section of the code can refer to [here](#current-license-data-structure-design-in-trustify)    
+
+ 2. Generate the CSV file and compress it. 
+    - This code can refer version one's code.
+ 3. Endpoint
+    - HTTP GET **/api/v2/sbom/{id}/license** Id: sbom id
+    
+#### UI
+License export action on the SBOM List and SBOM Detail screens.
+
+## Alternative approaches
+Currently, there is an alternative method. Instead of retrieving relevant information through the database's Entity, it directly obtains the original SBOM file (which already exists in the sbom_file Entity), parses it, and extracts all the information to generate the corresponding CSV files.
```

Comment:
The `sbom_file` entity **does not** contain SBOM files. It does contain the "file" elements from a SPDX SBOM. That's difference.

Also, I don't think this alternative exists at the moment. We could implement it though.

Reply:
I think [1] can got the resource file.
[1] https://github.com/trustification/trustify/blob/52fce44f662cd90729a031cdd1fdd6728ff3b473/modules/fundamental/src/sbom/endpoints/mod.rs#L399

Reply:
We do have the file stored. There's an internal method for retrieving it. I don't think we need to go into those details. However, I'd also prefer not having false information in the ADR.

Reply:
ok